lib/ipc_setup.c

```diff
@@ -865,3 +880,22 @@ qb_ipcs_us_connection_acceptor(int fd, int revent, void *data)
 	qb_ipcs_uc_recv_and_auth(new_fd, s);
 	return 0;
 }
+
+void remove_tempdir(const char *name, size_t namelen)
+{
+#if defined(QB_LINUX) || defined(QB_CYGWIN)
+	char dirname[PATH_MAX];
+	char *slash;
+	memcpy(dirname, name, namelen);
```

Comment:
Wouldn't `QB_MIN(nameline, sizeof dirname - 1)` be safer instead of `namelen` here (to reserve the final byte for a manual NUL)? Or a straight `strlcpy()`? Having `namelen` passed in raises doubts whether `name` is expected to be zero-terminated or not. The current uses are safe, though, except for the `strncat()` in `lib/ipc_setup.c`, which can overwrite the `description` field with a single byte (the closing NUL) and consequently leave the field unterminated.

Reply:
Actually, that `strncat()` can overwrite 3 (the length of its `src` argument, because it's smaller than `CONNECTION_DESCRIPTION`.